ActionMenu/ActionMenu.cs

```diff
@@ -689,6 +700,29 @@ public static Menus SplitOverCrowdedMenus(Menus m, uint overcrowdedMenuValue = 7
             return item;
         }
 
+        private static readonly System.Text.RegularExpressions.Regex impulseRe = new System.Text.RegularExpressions.Regex(
+            @"\s*<impulse(?:=(?<duration>\d+(?:\.?\d*)))>$",
+            System.Text.RegularExpressions.RegexOptions.IgnoreCase | System.Text.RegularExpressions.RegexOptions.Compiled
+            );
+
+        private static (string control, float? duration) DetectImpulseToggle(CVRAdvancedSettingsEntry s)
+        {
+            if (s.machineName.EndsWith("Impulse")) return (control: "impulse", duration: null);
+            var match = impulseRe.Match(s.name);
+            if (!match.Success) return (control: "toggle", duration: null);
+            var durationStr = match.Groups["duration"].Value;
+            if (durationStr != string.Empty && float.TryParse(durationStr, out var duration))
+            {
+                return (control: "impulse", duration: duration);
+            }
+            return (control: "impulse", duration: null);
+        }
+
+        private static string StripMenuNameTags(string menuName)

```

Comment:
Since item s.machineName and s.name exist we don't need to strip/clean tags, users can specify the clean name directly. Unless I misunderstand something.

Reply:
As mentioned in the comment above, it's necessary to strip the annotations.